classes/components/UserCommentComponent.php

```diff
@@ -0,0 +1,133 @@
+     */
+    public function __construct(Submission $submission, Request $request)
+    {
+        $this->request = $request;
+        $this->submission = $submission;
+
+        // get all published  publication ids for this submission
+        $this->publishedPublicationIds = collect();
+        foreach ($submission->getPublishedPublications() as $publishedPublication) {
+            $this->publishedPublicationIds->add($publishedPublication->getId());
+        }
+
+
+        $this->commentsCountPerPublication = UserComment::withPublicationIds($this->publishedPublicationIds->all())
+            ->withIsApproved(true)
+            ->select('publication_id', DB::raw('count(*) as count'))
+            ->groupBy('publication_id')
+            ->pluck('count', 'publication_id')
+            ->toArray();
+
+        $this->allCommentsCount = array_sum($this->commentsCountPerPublication);
+    }
+
+    /**
+     * Get the locale keys to expose in the pkpUserComment component.
+     *
+     */
+    public function getLocaleKeys(): array
+    {
+        return [
+            'userComment.discussionClosed',
+            'userComment.awaitingApprovalNotice',
+            'userComment.report.reason',
+            'common.cancel',
+            'userComment.reportCommentBy',
+            'userComment.reportCommentByUserWithAffiliation',
+            'userComment.report',
+            'userComment.showMore',
+            'form.submit',
+            'userComment.login',
+            'userComment.commentOnThisPublication',
+            'userComment.versionWithCount',
+            'common.delete',
+            'userComment.deleteCommentConfirmation',
+            'userComment.deleteComment',
+            'userComment.addYourComment',
+            'userComment.allComments',
+            'userComment.comments',
+            'userComment.reportComment',
+        ];
+    }
+
+    /**
+     * Get the URL to the login page.
+     *
+     * @return string
+     */
+    public function getLoginUrl()
+    {
+        return $this->request->getDispatcher()
+            ->url(
+                $this->request,
+                Application::ROUTE_PAGE,
+                null,
+                'login',
+                null,
+                null,
+                ['source' => $this->request->getRequestPath()]
+            );
+    }
+
+    /**
+     * Get the configuration for the pkpUserComment component.
+     *
+     * @return array
+     */
+    public function getConfig()
+    {
+        return [
+            'publicationIds' => $this->publishedPublicationIds->sortDesc()->values(),
+            'latestPublicationId' => $this->submission->getCurrentPublication()->getId(),
+            'itemsPerPage' => Repo::userComment()->getPerPage(),
+            'loginUrl' => $this->getLoginUrl(),
+            'allCommentsCount' => $this->allCommentsCount,
+            'commentsCountPerPublication' => $this->commentsCountPerPublication,
+        ];
```

Comment:
Still wondering, whether we could just have the good enough API to get most of these things from the API. 

Specifically  publicationIds, latestPublicationId, allCommentsCount and commentsCountPerPublication are very data oriented and something thats good fit for API.  The other ones are more matter of configuration.. so having that option to configure these server side is good convenience, but it would not prevent someone to implement this and just rely on the API.

@ewhanson @asmecher I understand we might want to take some shortcuts because of the timelines or current lack of the front office APIs. But would really appreciate your comment on this, whether you care about this or not.

Reply:
I don't have super well formed opinions about this, but on further reflection, I think you're right -- we should be careful about loading down `getConfig()` with things that should be in the API. It's a half-step away from our old server-side rendering patterns and we should be cleanly moving towards building out the API with the new VueJS stuff.